### PR TITLE
tests: re-enable pretty-std-collections on macOS

### LINKED ISSUE #78665: `debuginfo/pretty-std-collections.rs` test sometimes fails on macOS
CI on macOS sometimes fails due to this failure:
```
failures:

---- [debuginfo-lldb] debuginfo/pretty-std-collections.rs stdout ----
NOTE: compiletest thinks it is using LLDB version 1200
NOTE: compiletest thinks it is using LLDB without native rust support

error: Error while running LLDB
stderr:
------------------------------------------
error: need to add support for DW_TAG_base_type '()' encoded with DW_ATE = 0x7, bit_size = 0
clang: CommandLine Error: Option 'h' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options

------------------------------------------

failures:
    [debuginfo-lldb] debuginfo/pretty-std-collections.rs
```

You can find the full log here: https://github.com/rust-lang-ci/rust/runs/1340649828
The failure occurs on #78501, #78489, and #78661 (at least).

### PR DESCRIPTION
**Pull request by davidtwco**
Fixes #78665.

I made some small modifications to this test so that it would pass for me locally (though I was only able to test using lldb without built-in Rust support, but that seems to be the mode in which it would fail). I ran it a few hundred times with stage one and stage two to see if I could re-produce the spurious failures that were being reported in #78665 and couldn't. From the discussion in #78665, it seemed like this was related to Xcode versions and could be reproduced locally fairly easily. It's been a couple years since this was disabled so a lot has changed. If this starts failing spuriously again then we can disable it and I can look into that.

r? @wesleywiser (discussed in wg-debugging's triage meeting)

### CODE REVIEW

File: tests/debuginfo/pretty-std-collections.rs [hunk 1]
```diff
@@ -60,19 +59,19 @@
...
+// lldbg-check:[...] size=7 { [0] = 2 [1] = 3 [2] = 4 [3] = 5 [4] = 6 [5] = 7 [6] = 8 }
 // lldbr-check:(alloc::collections::vec_deque::VecDeque<i32>) vec_deque2 = size=7 = { [0] = 2 [1] = 3 [2] = 4 [3] = 5 [4] = 6 [5] = 7 [6] = 8 }
 
 // lldb-command:print hash_map
-// lldbg-check:[...]$2 = size=4 { [0] = { 0 = 1 1 = 10 } [1] = { 0 = 2 1 = 20 } [2] = { 0 = 3 1 = 30 } [3] = { 0 = 4 1 = 40 } }
+// lldbg-check:[...] size=4 { [0] = { 0 = 1 1 = 10 } [1] = { 0 = 2 1 = 20 } [2] = { 0 = 3 1 = 30 } [3] = { 0 = 4 1 = 40 } }
 // lldbr-check:(std::collections::hash::map::HashMap<u64, u64, [...]>) hash_map = size=4 size=4 { [0] = { 0 = 1 1 = 10 } [1] = { 0 = 2 1 = 20 } [2] = { 0 = 3 1 = 30 } [3] = { 0 = 4 1 = 40 } }
 
 // lldb-command:print hash_set
-// lldbg-check:[...]$3 = size=4 { [0] = 1 [1] = 2 [2] = 3 [3] = 4 }
```
Comment (davidtwco) [inline tests/debuginfo/pretty-std-collections.rs:75]: I've just now noticed - in working on another PR - that a lot of my debuginfo tests are failing requiring a change like this. I wonder if it is a lldb version issue or something else?

Comment (davidtwco) [reply tests/debuginfo/pretty-std-collections.rs:75]: Okay, I've confirmed this is a version issue, when using Xcode 13.4.1 (which is what CI uses, as best I can gather) then the test passes with the `$N =` parts, and when using Xcode 15.0 Beta 7 then it fails with these parts missing. I'll take a look and see if I can re-enable those with an lldb option and add that to compiletest.

Comment (davidtwco) [reply tests/debuginfo/pretty-std-collections.rs]: Sorted now with a change to compiletest.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
Nice! Thanks for looking into the lldb behavior change

@bors r+

**Comment by davidtwco** [comment]
So I had originally fixed the failures that happened with new lldb versions by aliasing `print` to `dwim-print --persistent-result on --` (lldb changed `print` from `expr --` to `dwim-print --` in a recent version, which had persistent results off by default). Instead I've just changed it back to `expr --` which hopefully will work in the older version we have on CI (otherwise I'll condition this alias on lldb version), and it also fixes the issue I was experiencing in llvm/llvm-project#65076.

**Review by wesleywiser — APPROVED** [review]

**Pull request merged**